tools/deleteorphanlogcommand.class.php

```diff
@@ -0,0 +1,193 @@
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Helper\ProgressBar;
+
+
+
+class DeleteOrphanLogCommand extends AbstractCommand {
+
+   protected $requires_db_up_to_date = false;
+
+   protected function configure() {
+      parent::configure();
+
+      $this->setName('glpi:tools:delete_orphan_logs');
+      $this->setAliases(
+         [
+            'tools:delete_orphan_logs',
+         ]
+      );
+      $this->setDescription(__('Delete orphan log'));
+
+      $this->addOption(
+         'dry-run', '',
+         InputOption::VALUE_NONE,
+         __('Only print the results whitout effective deletion')
+      );
+
+   }
+
+   protected function execute(InputInterface $input, OutputInterface $output) {
+
+      $no_interaction = $input->getOption('no-interaction');
+      if (!$no_interaction) {
+         // Ask for confirmation (unless --no-interaction)
+         $output->writeln([
+               __('You are about to delete orphan logs of GLPI log table (glpi_logs).'),
+               __('The first step check if orphans exists in database.'),
+               __('The final step is the definitive deletion of orphans (unless if --dry-run arguments is used used).')
+            ]
+         );
+
+         /**
+          * @var QuestionHelper $question_helper
+          */
+         $question_helper = $this->getHelper('question');
+         $run = $question_helper->ask(
+            $input,
+            $output,
+            new ConfirmationQuestion(
+               '<comment>' . __('Do you want to launch operation ?') . ' [yes/No]</comment>',
+               false
+            )
+         );
+         if (!$run) {
+            $output->writeln(
+               '<comment>' . __('Operation aborted.') . '</comment>',
+               OutputInterface::VERBOSITY_VERBOSE
+            );
+            return 0;
+         }
+      }
+
+      return $this->deleteOrphan($input, $output);
+   }
+
+   private function deleteOrphan(InputInterface $input, OutputInterface $output): bool {
+
+      $globalCount = 0;
+      $listTable = $this->db->listTables();
+      $dry_run = $this->input->getOption('dry-run');
+
+      if ($dry_run) {
+         $output->writeln(
+            '<comment>' . __('Effective deletion disabled (remove --dry-run arg)') . '</comment>'
+         );
+      }
+
+      $output->writeln(
+         '<comment>' . __('Check if orphans exists.') . '</comment>'
+      );
+
+      while ($table = $listTable->next()) {
+         $tablename = $table['TABLE_NAME'];
+         try {
+            $itemtype = getItemTypeForTable($tablename);
+         } catch (\Throwable $th) {
+            $output->writeln(
+               sprintf(
+                  '<error>' . __('Skip tablename %s : ERROR -> %s') . '</error>',
+                  $tablename,
+                  $th->getMessage()
+               )
+            );
+            continue;
+         }
+
```

Comment:
As this step can be long, using the present participle with ending `...` could help user to understand that process is working.
```suggestion
         $output->writeln(
            '<comment>' . sprintf(__('Searching for orphaned "%s"...'), $itemtype) . '</comment>',
            OutputInterface::VERBOSITY_QUIET
         );

```

Reply:
done